src/Report/Handler/Email/Strategy/AboveLevelWithContext.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Jh\Import\Report\Handler\Email\Strategy;
+
+use DateTime;
+use Jh\Import\LogLevel;
+use Jh\Import\Report\Message;
+use Jh\Import\Report\Report;
+
+class AboveLevelWithContext implements EmailHandlerStrategy
```

Comment:
This one's not finished, at least the rendering anyway. I've included it anyway as the logic is sound and should be finished at some point.